ibm/resource_ibm_dl_gateway.go

```diff
@@ -541,10 +542,20 @@ func resourceIBMdlGatewayCreate(d *schema.ResourceData, meta interface{}) error
 
 	log.Printf("[INFO] Created Direct Link Gateway (%s Template) : %s", dtype, *gateway.ID)
 	if dtype == "connect" {
-		_, err = isWaitForDirectLinkAvailable(directLink, d.Id(), d.Timeout(schema.TimeoutCreate))
+		getPortOptions := directLink.NewGetPortOptions(*gateway.Port.ID)
+		port, response, err := directLink.GetPort(getPortOptions)
 		if err != nil {
```

Comment:
if err != nil  || response != nil || port != nil

so that we are safe to handle the condition if port is nil by any chance